app/objects/impersonation/verifier.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Impersonation
+  class Verifier
+    EXPIRATION = 5.minutes
+    PURPOSE = :impersonation
+
+    def sign!(data)
+      data.then {
+        message_verifier.generate(_1, expires_in: EXPIRATION, purpose: PURPOSE)
+      }.then {
+        Base64.urlsafe_encode64(_1)
```

Comment:
Is it possible to replace`_1` with `it`?

Reply:
not yet, we need to wait til Christmas when Ruby 3.4 is going to be released 😅

Reply:
I ended up removing the `then` chain now that we have only 2 steps